playwright/_impl/_local_utils.py

```diff
@@ -26,3 +28,45 @@ def __init__(
 
     async def zip(self, zip_file: str, entries: List[NameValue]) -> None:
         await self._channel.send("zip", {"zipFile": zip_file, "entries": entries})
+
+    async def har_open(self, file: str) -> None:
+        params = locals_to_params(locals())
+        await self._channel.send("harOpen", params)
+
+    async def har_lookup(
+        self,
+        harId: str,
+        url: str,
+        method: str,
+        headers: HeadersArray,
+        isNavigationRequest: bool,
+        postData: Optional[bytes] = None,
+    ) -> HarLookupResult:
+        params = locals_to_params(locals())
+        if "postData" in params:
+            params["postData"] = base64.b64encode(params["postData"]).decode()
+        return cast(
+            HarLookupResult,
+            await self._channel.send_return_as_dict("harLookup", params),
+        )
+
+    async def har_close(self, harId: str) -> None:
+        params = locals_to_params(locals())
+        await self._channel.send("harClose", params)
+
+    def log(self, level: str, method: str, message: str) -> None:
+        try:
+            self._channel._connection.wrap_api_call(
+                lambda: self._channel.send_no_reply(
+                    method,
+                    {
+                        level: {
+                            "phase": "log",
+                            "message": message,
+                        },
+                    },
+                ),
+                True,
+            )
+        except Exception:
+            pass
```

Comment:
@mxschmitt Please advise on the recommended way for logging in this repo. This is based off of the code in the `_wait_helper.py`, but I suspect `method` cannot be an arbitrary string. As of now, I'm calling it like so elsewhere:

```
          self._local_utils.log(
                "debug",
                "api",
                f"HAR: {route.request.url} redirected to {redirect_url}",
            )
```

but I suspect it's wrong even though I see some output when piping through grep.

Reply:
We don't have logging via RPC, lets just not log or use the logging module? This way seems wrong and I doubt it will work.